app/library/zencart/DashboardWidget/src/WidgetManager.php

```diff
@@ -10,264 +10,284 @@
+        }
+
+        return $widgetList;
+    }
+
+
+    public function getWidgetsForUser()
+    {
+        $model = $this->modelFactory->make('DashboardWidgetsToUsers');
+        $widgets = $model->getWidgetInfoForUser($this->adminUser->admin_id)->toArray();
+        $widgetList = array();
+        foreach ($widgets as $widget) {
+            $widget['widget_name'] = $widget['dashboard_widget']['widget_name'];
+            $widgetList[$widget['widget_key']] = $widget;
+        }
+        return $widgetList;
     }
-    return $widgets;
-  }
-
-  public static function setWidgetTitle($name) {
-    if (defined($name)) $name = constant($name);
-    return $name;
-  }
-  public static function setWidgetDescription($name) {
-    if (defined($name."_DESCRIPTION")) {
-        $desc = constant($name."_DESCRIPTION");
-    } else {
-        $desc = "";
+
+
+    public function getWidgetTitle(string $name)
+    {
+        if (defined($name)) $name = constant($name);
+        return $name;
     }
-    return $desc;
-  }
-
-  public static function getWidgetInfoForUser($user)
-  {
-    global $db;
-    $sql = "SELECT * FROM " . TABLE_DASHBOARD_WIDGETS_TO_USERS . " as tdwtu
-            LEFT JOIN " . TABLE_DASHBOARD_WIDGETS_DESCRIPTION . " as tdwd ON tdwd.widget_key = tdwtu.widget_key
-            WHERE tdwtu.admin_id = :adminId:
-            ORDER BY widget_column, widget_row";
-    $sql = $db->bindVars($sql, ':adminId:', $user, 'integer');
-    $result = $db->execute($sql);
-    $widgets = array();
-    while (!$result->EOF)
+
+    public function getSettingsTitle(string $widgetName, string $settingsName)
     {
-      $widgets[$result->fields['widget_column']][$result->fields['widget_row']] = $result->fields;
-      $result->moveNext();
+        if (defined($widgetName . '_' . $settingsName)) {
+            $name = constant($widgetName . '_' . $settingsName);
+            return $name;
+        }
+        if (defined($settingsName)) {
+            $name = constant($settingsName);
+            return $name;
+        }
+        return 'UNDEFINED';
     }
-    return $widgets;
-  }
-
-  /**
-   * @param  array $widgetList
-   * @return array
-   * @todo   Is this necessary or valid with the new autoloading?
-   */
-  public static function loadWidgetClasses(array $widgetInfoList)
-  {
-    $widgetList = array();
-
-    foreach ($widgetInfoList as $widgets) {
-      foreach ($widgets as $widget) {
-        $classNameSpace = __NAMESPACE__ . '\\';
-        $className = base::camelize(strtolower(str_replace('-', '_', $widget['widget_key'])), TRUE);
-        $classFile = __DIR__ . DIRECTORY_SEPARATOR . $className . '.php';
-        $className = $classNameSpace . $className;
-
-        if (!class_exists($className, true) && is_readable($classFile)) {
-          require_once($classFile);
+
+
+    public function getWidgetDescription(string $name)
+    {
+        if (defined($name . "_DESCRIPTION")) {
+            $desc = constant($name . "_DESCRIPTION");
```

Comment:
Should we be using `strtoupper($name)`?

Reply:
noted, as mentioned in commit notes, I do want to further refactor the WidgetManager code. That function could also benefit from removing the else clause.